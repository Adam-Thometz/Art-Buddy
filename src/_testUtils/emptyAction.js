const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION