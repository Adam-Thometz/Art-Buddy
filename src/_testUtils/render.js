import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
