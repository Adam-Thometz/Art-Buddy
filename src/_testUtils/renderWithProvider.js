import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;