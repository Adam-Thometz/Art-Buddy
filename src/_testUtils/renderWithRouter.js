import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;