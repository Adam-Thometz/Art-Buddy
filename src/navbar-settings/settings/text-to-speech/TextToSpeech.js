import React from 'react';

import { useSelector, useDispatch } from 'react-redux';
import { toggleTextToSpeech } from '_redux/settings/mainSettingsActions';

import './TextToSpeech.css';

import Toggle from '_components/toggle/Toggle';

const TextToSpeech = () => {
  const { textToSpeech } = useSelector(state => state.mainSettings);
  const dispatch = useDispatch();
  
  const handleTextToSpeech = () => {
    dispatch(toggleTextToSpeech());
  };

  return (
    <div className='TextToSpeech'>
      <Toggle currToggle={textToSpeech} toggleFn={handleTextToSpeech} />
    </div>
  );
};

export default TextToSpeech;