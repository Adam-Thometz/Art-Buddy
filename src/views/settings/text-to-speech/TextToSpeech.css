.TextToSpeech {
  display: flex;
  align-items: center;
}
