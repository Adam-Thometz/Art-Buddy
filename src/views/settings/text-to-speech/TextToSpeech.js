import { useSelector, useDispatch } from "react-redux";
import { toggleTextToSpeech } from "store/settings/settingsReducer";

import "./TextToSpeech.css";

import Toggle from "components/toggle/Toggle";

const TextToSpeech = () => {
  const { textToSpeech } = useSelector((state) => state.settings);
  const dispatch = useDispatch();

  const handleTextToSpeech = () => dispatch(toggleTextToSpeech());

  return (
    <div className="TextToSpeech">
      <Toggle currToggle={textToSpeech} toggleFn={handleTextToSpeech} />
    </div>
  );
};

export default TextToSpeech;
