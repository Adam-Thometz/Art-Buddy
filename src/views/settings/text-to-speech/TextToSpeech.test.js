import TextToSpeech from "./TextToSpeech";

import { render } from "_testUtils/render";

describe("TextToSpeech component", () => {
  it("renders without crashing", () => {
    render(<TextToSpeech />);
  });

  it("matches the snapshot of TextToSpeech", () => {
    const { asFragment } = render(<TextToSpeech />);
    expect(asFragment()).toMatchSnapshot();
  });
});
