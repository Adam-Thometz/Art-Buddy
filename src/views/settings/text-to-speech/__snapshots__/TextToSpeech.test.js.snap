// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextToSpeech component matches the snapshot of TextToSpeech 1`] = `
<DocumentFragment>
  <div
    class="TextToSpeech"
  >
    <div
      class="Toggle"
    >
      <div
        class="Toggle-switch off"
      >
        <div
          class="Toggle-button"
        />
        <span>
          ON
        </span>
        <span>
          OFF
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
